src/server/dfly_main.cc

```diff
@@ -438,6 +439,114 @@ bool ShouldUseEpollAPI(const base::sys::KernelVersion& kver) {
   return true;
 }
 
+string get_cgroup_path(void) {
+  // Begin by reading /proc/self/cgroup
+
+  auto cg = io::ReadFileToString("/proc/self/cgroup");
+  CHECK(cg.has_value());
+
+  // Here, we assume that CGroup v1 is being used. This
+  // is quite likely, as CGroup v1 was introduced back in 2015.
+
+  // strip 0::<path> into just <path>, and newline.
+  auto cgv = std::move(cg.value());
+  return cgv.substr(3, cgv.length() - 3 - 1);
+  /**
+   * -3: we are skipping the first three characters
+   * -1: we discard the last character (a newline)
+   */
+}
+
+const auto cgroup = "/sys/fs/cgroup/" + get_cgroup_path();
+
+bool InsideContainer() {
+  /**
+   * (attemps) to check whether we are running
+   * inside a container or not.
+   *
+   * We employ several tests, all of which
+   * are flawed, however together do cover a
+   * good portion of cases.
+   *
+   * 1. Checking '/.dockerenv': very simple, however
+   * only works for Docker, and this file may be moved
+   * in the future.
+   * 2. Checking '/sys/fs/cgroup/memory.max': This
+   * directory -cannot- be edited (not even using sudoedit),
+   * so users are not able to add files to there. The file
+   * 'memory.max' should contain a memory limit for processes.
+   *
+   * We also use this file to find out how much memory we can use.
+   * However, on LXC this is placed in a different directory:
+   * 3. Checking '/sys/fs/cgroup/memory/memory.limit_in_bytes':
+   * Same idea.
+   */
+
+  using io::Exists;
+
+  return Exists("/.dockerenv") || Exists(cgroup + "/memory.max");
```

Comment:
why checking for .dockerenv? It's not enough to check for `memory.max`?

Reply:
I guess I left that check before adding the `cgroup` variable, it is redundant now.